test/sharness/lib/test-lib.sh

```diff
@@ -45,7 +45,9 @@ test "$TEST_EXPENSIVE" = 1 && test_set_prereq EXPENSIVE
 test "$TEST_NO_DOCKER" != 1 && type docker && test_set_prereq DOCKER
 
 TEST_OS=$(uname -s | tr [a-z] [A-Z])
-test expr "$TEST_OS" : "CYGWIN_NT" >/dev/null && test_set_prereq POSIX
+
+# Set a prereq as error messages are often different on Windows/Cygwin
+expr "$TEST_OS" : "CYGWIN_NT" >/dev/null || test_set_prereq STD_ERR_MSG
```

Comment:
Which platforms are "Linux-like" though? If we don't know and can't quantify it (are BSDs?), then we probably shouldn't claim that it's so.

> Also I like that it is specific to errors because when differences are not specific to errors we should not use a prereq but instead do something like what we do in do_umount()

Agreed. Since these *are* just platforms checks ("are we NOT cygwin?"), let's just use `if` statements like your example above. This prereq seems like an unnecessary semantic abstraction if all we're doing is wanting to know if we're on cygwin or not.